examples/plugins/upstreamauthority/plugin.go

```diff
@@ -0,0 +1,154 @@
+}
+
+func builtin(p *Plugin) catalog.Plugin {
+	return catalog.MakePlugin(pluginName, upstreamauthority.PluginServer(p))
+}
+
+type Config struct {
+	// TODO: fill in configurables you want to be able to control from the
+	// HCL configuration file.
+	//
+	// For example,
+	// SomeValue string `hcl:"some_value"`
+}
+
+type Plugin struct {
+	// mu is a mutex that protects the configuration. Plugins may at some point
+	// need to support hot-reloading of configuration (by receiving another
+	// call to Configure). So we need to prevent the configuration from
+	// being used concurrently and make sure it is updated atomically.
+	mu sync.Mutex
+	c  *Config
+}
+
+// These are compile time assertions that the plugin matches the interfaces the
+// catalog requires to provide the plugin with a logger and host service
+// broker as well as the UpstreamAuthority itself.
+var _ catalog.NeedsLogger = (*Plugin)(nil)
+var _ catalog.NeedsLogger = (*Plugin)(nil)
+var _ upstreamauthority.UpstreamAuthorityServer = (*Plugin)(nil)
+
+func New() *Plugin {
+	return &Plugin{}
+}
+
+// SetLogger will be called by the catalog system to provide the plugin with
+// a logger when it is loaded. The logger is wired up to the SPIRE core
+// logger
+func (p *Plugin) SetLogger(log hclog.Logger) {
+	// TODO: store the logger for later use. If the plugin does not need to
+	// log, this method can be removed.
+}
+
+// BrokerHostServices is called by the catalog system when the plugin is loaded
+// to provide it with host services implemented by SPIRE core.
+func (p *Plugin) BrokerHostServices(broker catalog.HostServiceBroker) error {
+	// TODO: use the broker to obtain clients to host services needed by the
+	// plugin for later use. If the plugin does not need any host services,
+	// this method can be removed.
+	return nil
+}
+
+// Mints an X.509 CA and responds with the signed X.509 CA certificate
+// chain and upstream X.509 roots. If supported by the implementation,
+// subsequent responses on the stream contain upstream X.509 root updates,
+// otherwise the RPC is completed after sending the initial response.
+//
+// Implementation note:
+// The stream should be kept open open in the face of transient errors
+// encountered while tracking changes to the upstream X.509 roots as SPIRE
+// core will not reopen a closed stream until the next X.509 CA rotation.
+func (p *Plugin) MintX509CA(req *upstreamauthority.MintX509CARequest, stream upstreamauthority.UpstreamAuthority_MintX509CAServer) error {
+	// TODO: implement
+	return nil
+}
+
+// Publishes a JWT signing key upstream and responds with the upstream JWT
+// keys. If supported by the implementation, subsequent responses on the
+// stream contain upstream JWT key updates, otherwise the RPC is completed
+// after sending the initial response.
+//
+// This RPC is optional and will return NotImplemented if unsupported.
+//
+// Implementation note:
+// The stream should be kept open open in the face of transient errors
```

Comment:
```suggestion
// The stream should be kept open in the face of transient errors
```

Reply:
This is also in the upstreamauthority.proto. I'll fix both.